src/main/java/com/hivemq/mqtt/handler/connect/ConnectPersistenceUpdateHandler.java

```diff
@@ -143,6 +147,7 @@ private void sendClientDisconnect(final Channel channel, final String clientId,
                     public void onSuccess(@Nullable final Void result) {
                         if (!channel.attr(ChannelAttributes.TAKEN_OVER).get()) {
                             channelPersistence.remove(clientId);
+                            metricsHolder.getLastWillTestamentSentCounter().inc();
```

Comment:
This does not mean that a will has been sent, it only means the client has been disconnected at the persistence level. It does also count disconnects without will. Better count the metric in com.hivemq.persistence.clientsession.PendingWillMessages#sendWill